pkg/server/server.go

```diff
@@ -122,6 +126,14 @@ func (s *Server) handleNarinfo(w http.ResponseWriter, r *http.Request) {
 			CA: pathInfo.CA,
 		}
 
+		if s.narServeCompression != "none" {
+			if s.narServeCompression == "zstd" {
+				narInfo.URL = narInfo.URL + ".zst"
+			} else {
+				narInfo.URL = narInfo.URL + "." + s.narServeCompression
```

Comment:
This _probably_ doesn't matter, but why is zstd special-cased here when brotli and gzip aren't?

Is there a problem with the files being called `nar.zstd` instead of `nar.zst`? Then why are `nar.gzip` instead of `nar.gz` and `nar.brotli` instead of `nar.br` fine? Or am I totally misreading this code?

Reply:
Yeah, I wanted to have the files still be named like the extensions Nix chooses during upload.

I checked on the `decompressor` side, `bz2` and `gz` extensions also need to be special-cased there.

I guess I'll move more of this code to `pkg/server/compression`, and move the lookup logic there too.

Reply:
The latest revision of that PR moved things to `compressor.go` - see bdba9909b3b0400f2c8230890ef9406dd8d7eab0,